File: API/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace API.Models
{
    public class Customer
    {
        [Key]
        public int CustomerId { get; set; }
        [Required]
        public string CustomerName { get; set; }
        [Required]
        public string Address { get; set; }
        [Required]
        [Range(1000, 9999)]
        public int Zip { get; set; }
        [Required]
        public string Place { get; set; }

        public List<Transaction> Transactions { get; set; }
    }
}

File: react-app/src/components/interfaces/Customer.ts
export interface Customer {
    customerId: number,
    customerName: string,
    address: string,
    zip: number,
    place: string
}

export const emptyCustomer: Customer = {
    customerId: 0,
    customerName: "",
    address: "",
    zip: 0,
    place: ""
}